packages/block-library/src/table-of-contents/index.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * Server-side rendering of the `core/table-of-contents` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Extracts Heading (`core/heading`) blocks from the current post.
+ *
+ * @return array The list of Heading blocks.
+ */
+function block_core_table_of_contents_get_heading_blocks() {
+	$blocks = parse_blocks( gutenberg_get_post_from_context()->post_content );
```

Comment:
To clarify, regarding https://github.com/WordPress/gutenberg/pull/21234#issuecomment-611313837, you had tried this without success?

```suggestion
	global $post;
	$blocks = parse_blocks( $post->post_content );
```

Reply:
Yeah, I tried that and it didn't work. I also tried this and it didn't work either:
```suggestion
	$blocks = parse_blocks( get_post()->post_content );
```